### ISSUE TITLE
Fix Encoding - [closed]

### ISSUE THREAD

**Issue by adumasphi** [persalys/persalys#503]
In GitLab by @aurelieladier on Nov 5, 2018, 11:48

_Merges encoding -> master_

Fix #91

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 5, 2018, 11:49

fix #91

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 12, 2018, 10:51

rejected proposition using Python api:
--------------------------------------

```
String Tools::GetUTF8String(const String& str)
{
  InterpreterUnlocker iul;
  Py_ssize_t size = strlen(str.c_str());
  String enc = "cp1252"; // get system encoding
  const char* errors = "";
  PyObject* decode_text = PyUnicode_Decode(str.c_str(), size, enc.c_str(), errors);
  PyObject* enc_text = PyUnicode_AsEncodedString(decode_text, "utf-8", errors);
  String typeString = checkAndConvert< _PyBytes_, String >(enc_text);
  return typeString;
}
```

proposition to get system encoding:
------------------------------------
works fine on linux, but on Windows...

```
  ScopedPyObjectPointer pyth_module(PyImport_ImportModule("locale"));
  ScopedPyObjectPointer pyth_func(PyObject_GetAttrString(pyth_module.get(), "getpreferredencoding"));
  if (pyth_func.get() && PyCallable_Check(pyth_func.get()))
  {
    ScopedPyObjectPointer pyth_val(PyObject_CallFunctionObjArgs(pyth_func.get()));
    if (pyth_val.get())
    {
      enc = checkAndConvert< _PyString_, String >(pyth_val.get());
    }
  }
```

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 14, 2018, 16:39

changed title from **{-proposition f-}ix Encoding** to **{+F+}ix Encoding**

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 14, 2018, 16:39

changed the description

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 14, 2018, 16:42

added 11 commits

<ul><li>fefc6e5a...be1c087b - 8 commits from branch <code>master</code></li><li>3eb1e9ef - Add Tools::GetLocaleString to convert utf8 string to system encoding string</li><li>69591f62 - ImportedDesignOfExperiment/DataModel/OTStudy: fix encoding on</li><li>c4d3d9bd - Add test t_Encoding_std</li></ul>

[Compare with previous version](https://git.phimeca.com/ot-edf/otgui/merge_requests/222/diffs?diff_id=1852&start_sha=fefc6e5a167161092158fa38df53fd6e23de1ed7)

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 15, 2018, 10:35

added 3 commits

<ul><li>da3f6712 - Add Tools::GetLocaleString to convert utf8 string to system encoding string</li><li>7586dcc4 - ImportedDesignOfExperiment/DataModel/OTStudy: fix encoding on</li><li>fe8bb8a8 - Add test t_Encoding_std</li></ul>

[Compare with previous version](https://git.phimeca.com/ot-edf/otgui/merge_requests/222/diffs?diff_id=1856&start_sha=c4d3d9bdda231dabe1eab1d0f4fd302835499c5a)

**Comment by adumasphi** [comment]
In GitLab by @aurelieladier on Nov 15, 2018, 16:29

closed
